T/upstream roc rk3399 pc

levindu opened the pull request:
This is an attempt to make csc support of `roc-rk3399-pc' to Armbian .

Currently, the default and dev branch is tested.

The mainline ATF has to be used, otherwise, it hangs at U-Boot.

piter75 commented:
> The mainline ATF has to be used, otherwise, it hangs at U-Boot.

That's interesting as 1.30 Rockchip's blob should have the same fix that was applied to mainline ATF: [mainline change](https://github.com/ARM-software/arm-trusted-firmware/commit/0aad563c74807195cc7fe2208d17e2d889157f1e) [rockchip change](https://github.com/rockchip-linux/rkbin/commit/a78ce16b17b1fb5fe72d6e93f0cd2fa139570bc3).

@levindu does it reboot for you correctly with master ATF?

In general I don't mind the mainline ATF - I was using it for 2 months without problems on my Rock Pi 4 with u-boot v2019.10-rc3 but I had to use v2.1 with minor fix to make the reboot working well.

The other thing that comes to my mind is that features / bugfixes of the mainline ATF may be incomplete for rk3399. Rockchip does not do much of active development there AFAIK.

We should probably make a conditional in `config/sources/rk3399.conf`, leave other boards on blob with `roc-3399-pc` built from mainline ATF.
We can then reconsider moving all other boards to mainline with some further discussion in the future.

@igorpecovnik WDYT?

Tonymac32 commented:
A few other rk3399 boards are similarly hanging at/failing at u-boot.  I tried a few small things, but did not mess with the ATF.  I agree with @Piter75 for the direction, move boards over or away as needed.  I also have this board and mezzanine, so I can test/tweak.

piter75 commented on config/sources/rk3399.conf:
```diff
@@ -18,6 +18,13 @@ BOOTDELAY=0
 OVERLAY_PREFIX='rockchip'
 SERIALCON='ttyFIQ0:1500000'
 
+ATFSOURCE='https://github.com/ARM-software/arm-trusted-firmware'
```
Please make this and above `UBOOT_TARGET_MAP` conditional for `roc-rk3399-pc` similarly to what is done in [`rockchip64-dev` family](https://github.com/armbian/build/blob/5e01f1234ed8d62569fe3ba9047780ced2d3ac79/config/sources/rockchip64.conf#L19)

levindu commented:
@piter75  Really, reboot failed. It hanged saying:

    root@roc:~# reboot
    [   35.550833] reboot: Restarting system

piter75 commented:
Yeah. That's what I remembered.

I probably need to get myself one of the boards hanging at modern u-boot, like your does, as I am pretty sure they share the same issues and making one work would probably make them all work.

levindu commented:
@piter75  How can I use to investigate this situation?

PR is updated to make it conditional to use mainline ATF.
Also a new patch about regulator is added to the next kernel.

piter75 commented:
Could you switch to the blob and try to boot it with the latest version from this branch?
Add the following to defconfig for your board to make sure it prints the debug messages and share the boot output.

```
CONFIG_LOG=y
CONFIG_LOG_CONSOLE=y
CONFIG_TPL_LOG=y
CONFIG_TPL_LOG_CONSOLE=y
CONFIG_SPL_LOG=y
CONFIG_SPL_LOG_CONSOLE=y
CONFIG_LOG_MAX_LEVEL=7
# CONFIG_USE_TINY_PRINTF is not set
```

levindu commented:
It boots, with log attched below:
http://ix.io/218M/text

Actually, I find out that defining DEBUG in MMC is essential in loading the kernel.
If I remove zzz-debug-rockpro64.patch, it does not boot.
Then I add DEBUG in `drivers/mmc/mmc.c`, it works.

I also report the same behavior in Firefly-RK3288 in the U-Boot mailing list:
https://lists.denx.de/pipermail/u-boot/2019-October/387082.html

levindu commented:
Weird, I just find out another way to work without DEBUG in mmc: turn on the log with following patch

```diff
diff --git a/configs/roc-rk3399-pc_defconfig b/configs/roc-rk3399-pc_defconfig                                                                        
index 809f5222368e..cf51236c89e2 100644
--- a/configs/roc-rk3399-pc_defconfig
+++ b/configs/roc-rk3399-pc_defconfig
@@ -54,3 +54,9 @@ CONFIG_USB_ETHER_RTL8152=y
 CONFIG_USB_ETHER_SMSC95XX=y
 CONFIG_SPL_TINY_MEMSET=y
 CONFIG_ERRNO_STR=y
+CONFIG_LOG=y
+CONFIG_LOG_CONSOLE=y
+CONFIG_SPL_LOG=y
+CONFIG_SPL_LOG_CONSOLE=y
+CONFIG_LOG_MAX_LEVEL=7
+# CONFIG_USE_TINY_PRINTF is not set
```

`CONFIG_LOG_MAX_LEVEL=7` is a must.

piter75 commented:
That is why I pointed them to you ;)

I was just going to ask you to try exactly this setup - removing debug but leaving those configs in place ;).

Can you try one more scenario?
Removing the zzz... patch altogether but leaving those configs in place.


>> On 8 Nov 2019, at 03:42, Levin Du <notifications@github.com> wrote:
> ﻿
> Weird, I just find out another way to work without DEBUG in mmc: turn on the log with following patch
> 
> diff --git a/configs/roc-rk3399-pc_defconfig b/configs/roc-rk3399-pc_defconfig                                                                        
> index 809f5222368e..cf51236c89e2 100644
> --- a/configs/roc-rk3399-pc_defconfig
> +++ b/configs/roc-rk3399-pc_defconfig
> @@ -54,3 +54,9 @@ CONFIG_USB_ETHER_RTL8152=y
>  CONFIG_USB_ETHER_SMSC95XX=y
>  CONFIG_SPL_TINY_MEMSET=y
>  CONFIG_ERRNO_STR=y
> +CONFIG_LOG=y
> +CONFIG_LOG_CONSOLE=y
> +CONFIG_SPL_LOG=y
> +CONFIG_SPL_LOG_CONSOLE=y
> +CONFIG_LOG_MAX_LEVEL=7
> +# CONFIG_USE_TINY_PRINTF is not set
> CONFIG_LOG_MAX_LEVEL=7 is a must.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

levindu commented:
I did exactly what you mentioned already ;-)
The previous result was exactly the one by removing the zzz-*.patch and adding the LOG config.
But I cannot understand why that did the trick.

piter75 commented:
I missed your second message ;-) I am glad that we found a solution.

Are you still willing to divert and build bootloader for `roc-rk3399-pc` with open ATF or do you want to stay with the blob like the rest of the boards?

levindu commented:
IMO, blobless boot for rk3399 is preferable ;-)

Tonymac32 commented:
I agree, as long as we have the functionality we need

piter75 commented:
I am totally in favour of OSS - that's why I also [started working on that](https://github.com/piter75/armbian-build/commit/80f8e73992ed7ae1009b8dcc67c9dc7ca80f7905).
OTOH I also like to reboot my machines remotely from time to time ;-)

The are two issues with mainline ATF right now:
1. I think that somewhere along this [commit](https://github.com/ARM-software/arm-trusted-firmware/commit/c1185ffde17cf2fdf50aac172a0be9e2d7669fd0#diff-94a9f1c135591eedd6addbcc729f46fb) it stopped resetting the CPU.
1. The other one is that even before that commit (eg. v2.1) it was resetting the CPU but not powering up the power domains before the reset and it just hang after a reset.

I did not resolve the first issue but was able to reboot with v2.1 with a small patch lying around my userpatches for the second one so we can at least make v2.1 working. I can provide the patch if you want to try and make your board work reliably with OSS ATF. Oh, and using v2.1 would require [this patch](https://github.com/ARM-software/arm-trusted-firmware/commit/0aad563c74807195cc7fe2208d17e2d889157f1e) to make it work with u-boot v2019.10.

OTOH we can make it work now with the blob - to finish the ongoing task of moving rk3399 to u-boot v2019.10 and open another issue that we can use to track the work needed to move all the rk3399 boards to OSS ATF.

Tonymac32 commented:
@piter75 of course, that's why I put the condition on it.  I didn't know the ATF master would work at all, but reboot is a must, I had to fix that on Tinker board by re-enabling the 3.3V regulator for SD right before the system actually reboot

piter75 commented:
Let's get that merged @levindu
In my branch I will apply the patches to fix reboot issues in mainline ATF.
Please test it in my branch.

levindu commented:
OK, I'll test it, thanks~

levindu commented:
Great news! When I typed `reboot` in the console, the board paused for a while, and reboot finally.

Here is the full log after reboot:
http://ix.io/22jg

Thanks for your awesome work!

Tonymac32 commented:
@levindu

My roc-pc is hanging at "Starting Kernel".  Can you verify this behavior against your own board?

```
U-Boot TPL 2019.10-armbian (Nov 28 2019 - 20:47:00)
Trying to boot from BOOTROM
Returning to boot ROM...

U-Boot SPL 2019.10-armbian (Nov 28 2019 - 20:47:00 -0500)
Trying to boot from MMC1


U-Boot 2019.10-armbian (Nov 28 2019 - 20:47:00 -0500)

Model: Firefly ROC-RK3399-PC Board
DRAM:  3.9 GiB
MMC:   dwmmc@fe320000: 1, sdhci@fe330000: 0
Loading Environment from EXT4... Card did not respond to voltage select!
In:    serial@ff1a0000
Out:   serial@ff1a0000
Err:   serial@ff1a0000
Model: Firefly ROC-RK3399-PC Board
rockchip_dnl_key_pressed: adc_channel_single_shot fail!
Net:
Error: ethernet@fe300000 address not set.
eth-1: ethernet@fe300000
Hit any key to stop autoboot:  0
Card did not respond to voltage select!
switch to partitions #0, OK
mmc1 is current device
Scanning mmc 1:1...
Found U-Boot script /boot/boot.scr
2949 bytes read in 5 ms (575.2 KiB/s)
## Executing script at 00500000
Boot script loaded from mmc 1
137 bytes read in 5 ms (26.4 KiB/s)
8100825 bytes read in 348 ms (22.2 MiB/s)
20449792 bytes read in 869 ms (22.4 MiB/s)
72014 bytes read in 12 ms (5.7 MiB/s)
2698 bytes read in 9 ms (292 KiB/s)
Applying kernel provided DT fixup script (rockchip-fixup.scr)
## Executing script at 39000000
## Loading init Ramdisk from Legacy Image at 04000000 ...
   Image Name:   uInitrd
   Image Type:   AArch64 Linux RAMDisk Image (gzip compressed)
   Data Size:    8100761 Bytes = 7.7 MiB
   Load Address: 00000000
   Entry Point:  00000000
   Verifying Checksum ... OK
## Flattened Device Tree blob at 01f00000
   Booting using the fdt blob at 0x1f00000
   Loading Ramdisk to f5765000, end f5f1eb99 ... OK
   Loading Device Tree to 00000000f56eb000, end 00000000f5764fff ... OK

Starting kernel ...
```

levindu commented:
@Tonymac32  It takes time to bring to the login.

Tonymac32 commented:
... Well, it has been an hour, still nothing.

levindu commented:
Couldn't be that long, here just a few minutes. What's your build option? I'll build one to test here.

Tonymac32 commented:
That was my assumption as well, I navigated the armbian menu system, selected the roc-pc, bionic with desktop

levindu commented:
Last I check, 5.3.0-rc4-rk3399 is okay, now it bumps to 5.4.

I'm having a problem with the dreadfully slow network downloading of the kernel.
Therefore it'll be a late response from me.

piter75 commented:
I just checked with the latest master and it booted fine with both 5.4.0-rc1 and 5.4.1 [enabled here](https://github.com/armbian/build/pull/1649).
However I noticed that there is a substantial (almost 2MiB) `initrd` size difference between your and my image.

```
U-Boot TPL 2019.10-armbian (Nov 29 2019 - 21:59:14)
Trying to boot from BOOTROM
Returning to boot ROM...

U-Boot SPL 2019.10-armbian (Nov 29 2019 - 21:59:14 +0000)
Trying to boot from MMC1
NOTICE:  BL31: v2.2(release):a04808c-dirty
NOTICE:  BL31: Built : 21:59:08, Nov 29 2019


U-Boot 2019.10-armbian (Nov 29 2019 - 21:59:14 +0000)

Model: Firefly ROC-RK3399-PC Board
DRAM:  3.9 GiB
MMC:   dwmmc@fe320000: 1, sdhci@fe330000: 0
Loading Environment from EXT4... Card did not respond to voltage select!
In:    serial@ff1a0000
Out:   serial@ff1a0000
Err:   serial@ff1a0000
Model: Firefly ROC-RK3399-PC Board
rockchip_dnl_key_pressed: adc_channel_single_shot fail!
Net:
Error: ethernet@fe300000 address not set.
eth-1: ethernet@fe300000
Hit any key to stop autoboot:  0
switch to partitions #0, OK
mmc1 is current device
Scanning mmc 1:1...
Found U-Boot script /boot/boot.scr
2949 bytes read in 5 ms (575.2 KiB/s)
## Executing script at 00500000
Boot script loaded from mmc 1
186 bytes read in 4 ms (44.9 KiB/s)
6127606 bytes read in 264 ms (22.1 MiB/s)
20449792 bytes read in 867 ms (22.5 MiB/s)
72014 bytes read in 12 ms (5.7 MiB/s)
2698 bytes read in 8 ms (329.1 KiB/s)
Applying kernel provided DT fixup script (rockchip-fixup.scr)
## Executing script at 39000000
## Loading init Ramdisk from Legacy Image at 04000000 ...
   Image Name:   uInitrd
   Image Type:   AArch64 Linux RAMDisk Image (gzip compressed)
   Data Size:    6127542 Bytes = 5.8 MiB
   Load Address: 00000000
   Entry Point:  00000000
   Verifying Checksum ... OK
## Flattened Device Tree blob at 01f00000
   Booting using the fdt blob at 0x1f00000
   Loading Ramdisk to f5946000, end f5f1dfb6 ... OK
   Loading Device Tree to 00000000f58cc000, end 00000000f5945fff ... OK

Starting kernel ...


Armbian 19.11.3 Buster ttyS2

roc login: root
Password:
Last login: Fri Nov 29 22:20:48 UTC 2019 on ttyS2
 ____   ___   ____      ____  _  ____________ ___   ___        ____   ____
|  _ \ / _ \ / ___|    |  _ \| |/ /___ /___ // _ \ / _ \      |  _ \ / ___|
| |_) | | | | |   _____| |_) | ' /  |_ \ |_ \ (_) | (_) |_____| |_) | |
|  _ <| |_| | |__|_____|  _ <| . \ ___) |__) \__, |\__, |_____|  __/| |___
|_| \_\\___/ \____|    |_| \_\_|\_\____/____/  /_/   /_/      |_|    \____|

Welcome to Armbian Buster with Linux 5.4.0-rc1-rockchip64

System load:   0.43 0.13 0.05  	Up time:       0 min
Memory usage:  2 % of 3868MB 	IP:
CPU temp:      48°C
Usage of /:    2% of 29G


root@roc:~# uname -a
Linux roc 5.4.0-rc1-rockchip64 #19.11.3 SMP PREEMPT Fri Nov 29 22:01:53 UTC 2019 aarch64 GNU/Linux
root@roc:~#
```

levindu commented:
The latest master branch build with
```
BOARD=roc-rk3399-pc
BRANCH=dev
RELEASE=bionic
BUILD_DESKTOP=yes
```
, with earlyprintk turning on, boots fine here (SD boot).

The uinitrd has different size comparing bionic with buster.

```
root@roc:~# reboot
[  374.078807] reboot: Restarting system

U-Boot TPL 2019.10-armbian (Dec 02 2019 - 01:43:00)
Trying to boot from BOOTROM
Returning to boot ROM...

U-Boot SPL 2019.10-armbian (Dec 02 2019 - 01:43:00 +0000)
Trying to boot from MMC1
NOTICE:  BL31: v2.2(release):a04808c-dirty
NOTICE:  BL31: Built : 01:42:50, Dec  2 2019


U-Boot 2019.10-armbian (Dec 02 2019 - 01:43:00 +0000)

Model: Firefly ROC-RK3399-PC Board
DRAM:  3.9 GiB
MMC:   dwmmc@fe320000: 1, sdhci@fe330000: 0
Loading Environment from EXT4... Card did not respond to voltage select!
In:    serial@ff1a0000
Out:   serial@ff1a0000
Err:   serial@ff1a0000
Model: Firefly ROC-RK3399-PC Board
rockchip_dnl_key_pressed: adc_channel_single_shot fail!
Net:
Error: ethernet@fe300000 address not set.
eth-1: ethernet@fe300000
Hit any key to stop autoboot:  0
Card did not respond to voltage select!
switch to partitions #0, OK
mmc1 is current device
Scanning mmc 1:1...
Found U-Boot script /boot/boot.scr
2949 bytes read in 5 ms (575.2 KiB/s)
## Executing script at 00500000
Boot script loaded from mmc 1
239 bytes read in 4 ms (57.6 KiB/s)
8100777 bytes read in 347 ms (22.3 MiB/s)
20449792 bytes read in 869 ms (22.4 MiB/s)
72014 bytes read in 12 ms (5.7 MiB/s)
2698 bytes read in 8 ms (329.1 KiB/s)
Applying kernel provided DT fixup script (rockchip-fixup.scr)
## Executing script at 39000000
## Loading init Ramdisk from Legacy Image at 04000000 ...
   Image Name:   uInitrd
   Image Type:   AArch64 Linux RAMDisk Image (gzip compressed)
   Data Size:    8100713 Bytes = 7.7 MiB
   Load Address: 00000000
   Entry Point:  00000000
   Verifying Checksum ... OK
## Flattened Device Tree blob at 01f00000
   Booting using the fdt blob at 0x1f00000
   Loading Ramdisk to f5765000, end f5f1eb69 ... OK
   Loading Device Tree to 00000000f56eb000, end 00000000f5764fff ... OK

Starting kernel ...

[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd034]
[    0.000000] Linux version 5.4.0-rc1-rockchip64 (root@00ea993abfd6) (gcc version 8.3.0 (GNU Toolchain for the A-profile Architecture 8.3-2019.03 (arm-rel-8.36))) #19.11.3 SMP PREEMPT Mon Dec 2 01:49:54 UTC 2019
[    0.000000] Machine model: Firefly ROC-RK3399-PC Board
[    0.000000] earlycon: uart8250 at MMIO32 0x00000000ff1a0000 (options '')
[    0.000000] printk: bootconsole [uart8250] enabled
[    0.000000] efi: Getting EFI parameters from FDT:
[    0.000000] efi: UEFI not found.
[    0.000000] cma: Reserved 16 MiB at 0x00000000f7000000
[    0.000000] NUMA: No NUMA configuration found
[    0.000000] NUMA: Faking a node at [mem 0x0000000000200000-0x00000000f7ffffff]
[    0.000000] NUMA: NODE_DATA [mem 0xf67e5800-0xf67e6fff]
[    0.000000] Zone ranges:
[    0.000000]   DMA32    [mem 0x0000000000200000-0x00000000f7ffffff]
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000200000-0x00000000f7ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000200000-0x00000000f7ffffff]
[    0.000000] psci: probing for conduit method from DT.
[    0.000000] psci: PSCIv1.1 detected in firmware.
[    0.000000] psci: Using standard PSCI v0.2 function IDs
[    0.000000] psci: MIGRATE_INFO_TYPE not supported.
[    0.000000] psci: SMC Calling Convention v1.1
[    0.000000] percpu: Embedded 22 pages/cpu s52376 r8192 d29544 u90112
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] CPU features: detected: GIC system register CPU interface
[    0.000000] Speculative Store Bypass Disable mitigation not required
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 999432
[    0.000000] Policy zone: DMA32
[    0.000000] Kernel command line: root=UUID=ac4fb0bb-12da-4766-8500-903f71b51035 rootwait rootfstype=ext4 console=ttyS2,1500000 console=tty1 panic=10 consoleblank=0 loglevel=7 ubootpart=2e0a4401-01 usb-storage.quirks=0x2537:0x1066:u,0x2537:0x1068:u printk earlycon=uart8250,mmio32,0xff1a0000  cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory swapaccount=1
[    0.000000] Dentry cache hash table entries: 524288 (order: 10, 4194304 bytes, linear)
[    0.000000] Inode-cache hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 3935920K/4061184K available (13308K kernel code, 898K rwdata, 4776K rodata, 896K init, 388K bss, 108880K reserved, 16384K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=6, Nodes=1
[    0.000000] rcu: Preemptible hierarchical RCU implementation.
[    0.000000] rcu:     RCU restricting CPUs from NR_CPUS=8 to nr_cpu_ids=6.
[    0.000000]  Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 25 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=6
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] GICv3: GIC: Using split EOI/Deactivate mode
[    0.000000] GICv3: 988 SPIs implemented
[    0.000000] GICv3: 0 Extended SPIs implemented
[    0.000000] GICv3: Distributor has no Range Selector support
[    0.000000] GICv3: 16 PPIs implemented
[    0.000000] GICv3: no VLPI support, no direct LPI support
[    0.000000] GICv3: CPU0: found redistributor 0 region 0:0x00000000fef00000
[    0.000000] ITS [mem 0xfee20000-0xfee3ffff]
[    0.000000] ITS@0x00000000fee20000: allocated 65536 Devices @f1080000 (flat, esz 8, psz 64K, shr 0)
[    0.000000] ITS: using cache flushing for cmd queue
[    0.000000] GICv3: using LPI property table @0x00000000f1040000
[    0.000000] GIC: using cache flushing for LPI property table
[    0.000000] GICv3: CPU0: using allocated LPI pending table @0x00000000f1050000
[    0.000000] GICv3: GIC: PPI partition interrupt-partition-0[0] { /cpus/cpu@0[0] /cpus/cpu@1[1] /cpus/cpu@2[2] /cpus/cpu@3[3] }
[    0.000000] GICv3: GIC: PPI partition interrupt-partition-1[1] { /cpus/cpu@100[4] /cpus/cpu@101[5] }
[    0.000000] random: get_random_bytes called from start_kernel+0x2ac/0x438 with crng_init=0
[    0.000000] arch_timer: cp15 timer(s) running at 24.00MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x588fe9dc0, max_idle_ns: 440795202592 ns
[    0.000007] sched_clock: 56 bits at 24MHz, resolution 41ns, wraps every 4398046511097ns
[    0.002266] Console: colour dummy device 80x25
[    0.002699] printk: console [tty1] enabled
[    0.003105] printk: bootconsole [uart8250] disabled
[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd034]
[    0.000000] Linux version 5.4.0-rc1-rockchip64 (root@00ea993abfd6) (gcc version 8.3.0 (GNU Toolchain for the A-profile Architecture 8.3-2019.03 (arm-rel-8.36))) #19.11.3 SMP PREEMPT Mon Dec 2 01:49:54 UTC 2019
[    0.000000] Machine model: Firefly ROC-RK3399-PC Board
[    0.000000] earlycon: uart8250 at MMIO32 0x00000000ff1a0000 (options '')
[    0.000000] printk: bootconsole [uart8250] enabled
[    0.000000] efi: Getting EFI parameters from FDT:
[    0.000000] efi: UEFI not found.
[    0.000000] cma: Reserved 16 MiB at 0x00000000f7000000
[    0.000000] NUMA: No NUMA configuration found
[    0.000000] NUMA: Faking a node at [mem 0x0000000000200000-0x00000000f7ffffff]
[    0.000000] NUMA: NODE_DATA [mem 0xf67e5800-0xf67e6fff]
[    0.000000] Zone ranges:
[    0.000000]   DMA32    [mem 0x0000000000200000-0x00000000f7ffffff]
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000200000-0x00000000f7ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000200000-0x00000000f7ffffff]
[    0.000000] psci: probing for conduit method from DT.
[    0.000000] psci: PSCIv1.1 detected in firmware.
[    0.000000] psci: Using standard PSCI v0.2 function IDs
[    0.000000] psci: MIGRATE_INFO_TYPE not supported.
[    0.000000] psci: SMC Calling Convention v1.1
[    0.000000] percpu: Embedded 22 pages/cpu s52376 r8192 d29544 u90112
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] CPU features: detected: GIC system register CPU interface
[    0.000000] Speculative Store Bypass Disable mitigation not required
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 999432
[    0.000000] Policy zone: DMA32
[    0.000000] Kernel command line: root=UUID=ac4fb0bb-12da-4766-8500-903f71b51035 rootwait rootfstype=ext4 console=ttyS2,1500000 console=tty1 panic=10 consoleblank=0 loglevel=7 ubootpart=2e0a4401-01 usb-storage.quirks=0x2537:0x1066:u,0x2537:0x1068:u printk earlycon=uart8250,mmio32,0xff1a0000  cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory swapaccount=1
[    0.000000] Dentry cache hash table entries: 524288 (order: 10, 4194304 bytes, linear)
[    0.000000] Inode-cache hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 3935920K/4061184K available (13308K kernel code, 898K rwdata, 4776K rodata, 896K init, 388K bss, 108880K reserved, 16384K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=6, Nodes=1
[    0.000000] rcu: Preemptible hierarchical RCU implementation.
[    0.000000] rcu:     RCU restricting CPUs from NR_CPUS=8 to nr_cpu_ids=6.
[    0.000000]  Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 25 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=6
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] GICv3: GIC: Using split EOI/Deactivate mode
[    0.000000] GICv3: 988 SPIs implemented
[    0.000000] GICv3: 0 Extended SPIs implemented
[    0.000000] GICv3: Distributor has no Range Selector support
[    0.000000] GICv3: 16 PPIs implemented
[    0.000000] GICv3: no VLPI support, no direct LPI support
[    0.000000] GICv3: CPU0: found redistributor 0 region 0:0x00000000fef00000
[    0.000000] ITS [mem 0xfee20000-0xfee3ffff]
[    0.000000] ITS@0x00000000fee20000: allocated 65536 Devices @f1080000 (flat, esz 8, psz 64K, shr 0)
[    0.000000] ITS: using cache flushing for cmd queue
[    0.000000] GICv3: using LPI property table @0x00000000f1040000
[    0.000000] GIC: using cache flushing for LPI property table
[    0.000000] GICv3: CPU0: using allocated LPI pending table @0x00000000f1050000
[    0.000000] GICv3: GIC: PPI partition interrupt-partition-0[0] { /cpus/cpu@0[0] /cpus/cpu@1[1] /cpus/cpu@2[2] /cpus/cpu@3[3] }
[    0.000000] GICv3: GIC: PPI partition interrupt-partition-1[1] { /cpus/cpu@100[4] /cpus/cpu@101[5] }
[    0.000000] random: get_random_bytes called from start_kernel+0x2ac/0x438 with crng_init=0
[    0.000000] arch_timer: cp15 timer(s) running at 24.00MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x588fe9dc0, max_idle_ns: 440795202592 ns
[    0.000007] sched_clock: 56 bits at 24MHz, resolution 41ns, wraps every 4398046511097ns
[    0.002266] Console: colour dummy device 80x25
[    0.002699] printk: console [tty1] enabled
[    0.003105] printk: bootconsole [uart8250] disabled
[    0.003689] Calibrating delay loop (skipped), value calculated using timer frequency.. 48.00 BogoMIPS (lpj=96000)
[    0.003727] pid_max: default: 32768 minimum: 301
[    0.003869] LSM: Security Framework initializing
[    0.003982] Mount-cache hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.004030] Mountpoint-cache hash table entries: 8192 (order: 4, 65536 bytes, linear)
[    0.004119] *** VALIDATE tmpfs ***
[    0.004769] *** VALIDATE proc ***
[    0.005100] *** VALIDATE cgroup1 ***
[    0.005123] *** VALIDATE cgroup2 ***
[    0.026340] ASID allocator initialised with 32768 entries
[    0.034335] rcu: Hierarchical SRCU implementation.
[    0.042490] Platform MSI: interrupt-controller@fee20000 domain created
[    0.042950] PCI/MSI: /interrupt-controller@fee00000/interrupt-controller@fee20000 domain created
[    0.043273] EFI services will not be available.
[    0.050405] smp: Bringing up secondary CPUs ...
[    0.082642] Detected VIPT I-cache on CPU1
[    0.082680] GICv3: CPU1: found redistributor 1 region 0:0x00000000fef20000
[    0.082697] GICv3: CPU1: using allocated LPI pending table @0x00000000f1060000
[    0.082745] CPU1: Booted secondary processor 0x0000000001 [0x410fd034]
[    0.114731] Detected VIPT I-cache on CPU2
[    0.114762] GICv3: CPU2: found redistributor 2 region 0:0x00000000fef40000
[    0.114776] GICv3: CPU2: using allocated LPI pending table @0x00000000f1070000
[    0.114809] CPU2: Booted secondary processor 0x0000000002 [0x410fd034]
[    0.146845] Detected VIPT I-cache on CPU3
[    0.146873] GICv3: CPU3: found redistributor 3 region 0:0x00000000fef60000
[    0.146887] GICv3: CPU3: using allocated LPI pending table @0x00000000f1100000
[    0.146919] CPU3: Booted secondary processor 0x0000000003 [0x410fd034]
[    0.178974] CPU features: detected: EL2 vector hardening
[    0.178999] Detected PIPT I-cache on CPU4
[    0.179031] GICv3: CPU4: found redistributor 100 region 0:0x00000000fef80000
[    0.179044] GICv3: CPU4: using allocated LPI pending table @0x00000000f1110000
[    0.179081] CPU4: Booted secondary processor 0x0000000100 [0x410fd082]
[    0.211096] Detected PIPT I-cache on CPU5
[    0.211124] GICv3: CPU5: found redistributor 101 region 0:0x00000000fefa0000
[    0.211136] GICv3: CPU5: using allocated LPI pending table @0x00000000f1120000
[    0.211165] CPU5: Booted secondary processor 0x0000000101 [0x410fd082]
[    0.211306] smp: Brought up 1 node, 6 CPUs
[    0.211682] SMP: Total of 6 processors activated.
[    0.211704] CPU features: detected: 32-bit EL0 Support
[    0.211729] CPU features: detected: CRC32 instructions
[    0.212876] CPU: All CPU(s) started at EL2
[    0.212936] alternatives: patching kernel code
[    0.216160] devtmpfs: initialized
[    0.236439] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.236502] futex hash table entries: 2048 (order: 5, 131072 bytes, linear)
[    0.237562] xor: measuring software checksum speed
[    0.275175]    8regs     :  1188.000 MB/sec
[    0.315250]    32regs    :  2760.000 MB/sec
[    0.355326]    arm64_neon:  2368.000 MB/sec
[    0.355342] xor: using function: 32regs (2760.000 MB/sec)
[    0.355437] pinctrl core: initialized pinctrl subsystem
[    0.356271] DMI not present or invalid.
[    0.356632] NET: Registered protocol family 16
[    0.372598] DMA: preallocated 2048 KiB pool for atomic allocations
[    0.372624] audit: initializing netlink subsys (disabled)
[    0.372826] audit: type=2000 audit(0.372:1): state=initialized audit_enabled=0 res=1
[    0.374109] cpuidle: using governor menu
[    0.374536] hw-breakpoint: found 6 breakpoint and 4 watchpoint registers.
[    0.374848] Serial: AMBA PL011 UART driver
[    0.415494] HugeTLB registered 1.00 GiB page size, pre-allocated 0 pages
[    0.415522] HugeTLB registered 32.0 MiB page size, pre-allocated 0 pages
[    0.415540] HugeTLB registered 2.00 MiB page size, pre-allocated 0 pages
[    0.415558] HugeTLB registered 64.0 KiB page size, pre-allocated 0 pages
[    0.426344] cryptd: max_cpu_qlen set to 1000
[    0.507729] raid6: neonx8   gen()  1325 MB/s
[    0.575854] raid6: neonx8   xor()  1201 MB/s
[    0.643998] raid6: neonx4   gen()  1127 MB/s
[    0.712118] raid6: neonx4   xor()   995 MB/s
[    0.780277] raid6: neonx2   gen()  1017 MB/s
[    0.848436] raid6: neonx2   xor()  1062 MB/s
[    0.916554] raid6: neonx1   gen()   724 MB/s
[    0.984681] raid6: neonx1   xor()   801 MB/s
[    1.052843] raid6: int64x8  gen()   836 MB/s
[    1.120949] raid6: int64x8  xor()   531 MB/s
[    1.189106] raid6: int64x4  gen()   752 MB/s
[    1.257219] raid6: int64x4  xor()   550 MB/s
[    1.325422] raid6: int64x2  gen()   524 MB/s
[    1.393513] raid6: int64x2  xor()   497 MB/s
[    1.461635] raid6: int64x1  gen()   355 MB/s
[    1.529728] raid6: int64x1  xor()   322 MB/s
[    1.529742] raid6: using algorithm neonx8 gen() 1325 MB/s
[    1.529756] raid6: .... xor() 1201 MB/s, rmw enabled
[    1.529769] raid6: using neon recovery algorithm
[    1.530767] vcc12v_sys: supplied by vcc_vbus_typec0
[    1.531338] vcc3v3_sys: supplied by vcc12v_sys
[    1.532503] vcc_sys: supplied by vcc12v_sys
[    1.533016] iommu: Default domain type: Translated
[    1.533538] rk_iommu ff650800.iommu: IRQ index 1 not found
[    1.533906] rk_iommu ff8f3f00.iommu: IRQ index 1 not found
[    1.534262] rk_iommu ff903f00.iommu: IRQ index 1 not found
[    1.534656] rk_iommu ff914000.iommu: IRQ index 1 not found
[    1.534986] rk_iommu ff924000.iommu: IRQ index 1 not found
[    1.535275] vgaarb: loaded
[    1.535590] SCSI subsystem initialized
[    1.536058] usbcore: registered new interface driver usbfs
[    1.536112] usbcore: registered new interface driver hub
[    1.536206] usbcore: registered new device driver usb
[    1.536566] pps_core: LinuxPPS API ver. 1 registered
[    1.536582] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    1.536611] PTP clock support registered
[    1.536857] EDAC MC: Ver: 3.0.0
[    1.537438] Advanced Linux Sound Architecture Driver Initialized.
[    1.537898] NetLabel: Initializing
[    1.537914] NetLabel:  domain hash size = 128
[    1.537926] NetLabel:  protocols = UNLABELED CIPSOv4 CALIPSO
[    1.537998] NetLabel:  unlabeled traffic allowed by default
[    1.538605] clocksource: Switched to clocksource arch_sys_counter
[    1.538796] VFS: Disk quotas dquot_6.6.0
[    1.538870] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    1.538931] *** VALIDATE ramfs ***
[    1.538953] *** VALIDATE hugetlbfs ***
[    1.547586] thermal_sys: Registered thermal governor 'step_wise'
[    1.547591] thermal_sys: Registered thermal governor 'power_allocator'
[    1.548154] NET: Registered protocol family 2
[    1.548827] tcp_listen_portaddr_hash hash table entries: 2048 (order: 3, 32768 bytes, linear)
[    1.548926] TCP established hash table entries: 32768 (order: 6, 262144 bytes, linear)
[    1.549285] TCP bind hash table entries: 32768 (order: 7, 524288 bytes, linear)
[    1.549962] TCP: Hash tables configured (established 32768 bind 32768)
[    1.550095] UDP hash table entries: 2048 (order: 4, 65536 bytes, linear)
[    1.550243] UDP-Lite hash table entries: 2048 (order: 4, 65536 bytes, linear)
[    1.550559] NET: Registered protocol family 1
[    1.551203] RPC: Registered named UNIX socket transport module.
[    1.551222] RPC: Registered udp transport module.
[    1.551235] RPC: Registered tcp transport module.
[    1.551248] RPC: Registered tcp NFSv4.1 backchannel transport module.
[    1.551269] PCI: CLS 0 bytes, default 64
[    1.551441] Trying to unpack rootfs image as initramfs...
[    2.081702] Freeing initrd memory: 7908K
[    2.082958] hw perfevents: enabled with armv8_cortex_a53 PMU driver, 7 counters available
[    2.083336] hw perfevents: enabled with armv8_cortex_a72 PMU driver, 7 counters available
[    2.083784] kvm [1]: IPA Size Limit: 40bits
[    2.084407] kvm [1]: vgic-v2@fff20000
[    2.084451] kvm [1]: GIC system register CPU interface enabled
[    2.084670] kvm [1]: vgic interrupt IRQ10
[    2.084925] kvm [1]: Hyp mode initialized successfully
[    2.182691] Initialise system trusted keyrings
[    2.182889] workingset: timestamp_bits=44 max_order=20 bucket_order=0
[    2.190258] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    2.191065] NFS: Registering the id_resolver key type
[    2.191110] Key type id_resolver registered
[    2.191125] Key type id_legacy registered
[    2.191148] nfs4filelayout_init: NFSv4 File Layout Driver Registering...
[    2.191164] Installing knfsd (copyright (C) 1996 okir@monad.swb.de).
[    2.192212] fuse: init (API version 7.31)
[    2.192309] *** VALIDATE fuse ***
[    2.192329] *** VALIDATE fuse ***
[    2.192748] SGI XFS with ACLs, security attributes, realtime, no debug enabled
[    2.194260] 9p: Installing v9fs 9p2000 file system support
[    2.224509] NET: Registered protocol family 38
[    2.224540] Key type asymmetric registered
[    2.224556] Asymmetric key parser 'x509' registered
[    2.224618] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 247)
[    2.224827] io scheduler mq-deadline registered
[    2.224844] io scheduler kyber registered
[    2.225010] io scheduler bfq registered
[    2.227100] vcc5v0_host: supplied by vcc_sys
[    2.228941] vcc_vbus_typec1: supplied by vcc_sys
[    2.234194] dma-pl330 ff6d0000.dma-controller: Loaded driver for PL330 DMAC-241330
[    2.234223] dma-pl330 ff6d0000.dma-controller:       DBUFF-32x8bytes Num_Chans-6 Num_Peri-12 Num_Events-12
[    2.235501] dma-pl330 ff6e0000.dma-controller: Loaded driver for PL330 DMAC-241330
[    2.235528] dma-pl330 ff6e0000.dma-controller:       DBUFF-128x8bytes Num_Chans-8 Num_Peri-20 Num_Events-16
[    2.237223] pwm-regulator: supplied by regulator-dummy
[    2.237931] Serial: 8250/16550 driver, 5 ports, IRQ sharing enabled
[    2.239772] ff180000.serial: ttyS0 at MMIO 0xff180000 (irq = 35, base_baud = 1500000) is a 16550A
[    2.240457] ff1a0000.serial: ttyS2 at MMIO 0xff1a0000 (irq = 36, base_baud = 1500000) is a 16550A
[    2.356424] printk: console [ttyS2] enabled
[    2.360703] cacheinfo: Unable to detect cache hierarchy for CPU 0
[    2.369422] brd: module loaded
[    2.379454] loop: module loaded
[    2.440915] libphy: Fixed MDIO Bus: probed
[    2.441771] tun: Universal TUN/TAP device driver, 1.6
[    2.443764] e1000e: Intel(R) PRO/1000 Network Driver - 3.2.6-k
[    2.444287] e1000e: Copyright(c) 1999 - 2015 Intel Corporation.
[    2.444854] igb: Intel(R) Gigabit Ethernet Network Driver - version 5.6.0-k
[    2.445473] igb: Copyright (c) 2007-2014 Intel Corporation.
[    2.446005] igbvf: Intel(R) Gigabit Virtual Function Network Driver - version 2.4.0-k
[    2.446721] igbvf: Copyright (c) 2009 - 2012 Intel Corporation.
[    2.447867] pegasus: v0.9.3 (2013/04/25), Pegasus/Pegasus II USB Ethernet driver
[    2.448561] usbcore: registered new interface driver pegasus
[    2.449104] usbcore: registered new interface driver rtl8150
[    2.449645] usbcore: registered new interface driver r8152
[    2.450164] usbcore: registered new interface driver dm9601
[    2.450715] usbcore: registered new interface driver CoreChips
[    2.451277] usbcore: registered new interface driver smsc75xx
[    2.451835] usbcore: registered new interface driver smsc95xx
[    2.452379] usbcore: registered new interface driver plusb
[    2.452904] usbcore: registered new interface driver MOSCHIP usb-ethernet driver
[    2.453733] VFIO - User Level meta-driver version: 0.3
[    2.457895] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    2.458486] ehci-pci: EHCI PCI platform driver
[    2.458956] ehci-platform: EHCI generic platform driver
[    2.461665] ehci-platform fe380000.usb: EHCI Host Controller
[    2.462193] ehci-platform fe380000.usb: new USB bus registered, assigned bus number 1
[    2.463019] ehci-platform fe380000.usb: irq 27, io mem 0xfe380000
[    2.478635] ehci-platform fe380000.usb: USB 2.0 started, EHCI 1.00
[    2.479343] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 5.04
[    2.480083] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    2.480730] usb usb1: Product: EHCI Host Controller
[    2.481170] usb usb1: Manufacturer: Linux 5.4.0-rc1-rockchip64 ehci_hcd
[    2.481761] usb usb1: SerialNumber: fe380000.usb
[    2.482701] hub 1-0:1.0: USB hub found
[    2.483074] hub 1-0:1.0: 1 port detected
[    2.485950] ehci-platform fe3c0000.usb: EHCI Host Controller
[    2.486474] ehci-platform fe3c0000.usb: new USB bus registered, assigned bus number 2
[    2.487292] ehci-platform fe3c0000.usb: irq 29, io mem 0xfe3c0000
[    2.502631] ehci-platform fe3c0000.usb: USB 2.0 started, EHCI 1.00
[    2.503322] usb usb2: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 5.04
[    2.504062] usb usb2: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    2.504708] usb usb2: Product: EHCI Host Controller
[    2.505147] usb usb2: Manufacturer: Linux 5.4.0-rc1-rockchip64 ehci_hcd
[    2.505739] usb usb2: SerialNumber: fe3c0000.usb
[    2.506617] hub 2-0:1.0: USB hub found
[    2.506987] hub 2-0:1.0: 1 port detected
[    2.507889] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
[    2.508462] ohci-pci: OHCI PCI platform driver
[    2.508915] ohci-platform: OHCI generic platform driver
[    2.509587] ohci-platform fe3a0000.usb: Generic Platform OHCI controller
[    2.510200] ohci-platform fe3a0000.usb: new USB bus registered, assigned bus number 3
[    2.511017] ohci-platform fe3a0000.usb: irq 28, io mem 0xfe3a0000
[    2.574782] usb usb3: New USB device found, idVendor=1d6b, idProduct=0001, bcdDevice= 5.04
[    2.575522] usb usb3: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    2.576169] usb usb3: Product: Generic Platform OHCI controller
[    2.576698] usb usb3: Manufacturer: Linux 5.4.0-rc1-rockchip64 ohci_hcd
[    2.577290] usb usb3: SerialNumber: fe3a0000.usb
[    2.578141] hub 3-0:1.0: USB hub found
[    2.578519] hub 3-0:1.0: 1 port detected
[    2.579345] ohci-platform fe3e0000.usb: Generic Platform OHCI controller
[    2.579957] ohci-platform fe3e0000.usb: new USB bus registered, assigned bus number 4
[    2.580757] ohci-platform fe3e0000.usb: irq 30, io mem 0xfe3e0000
[    2.642831] usb usb4: New USB device found, idVendor=1d6b, idProduct=0001, bcdDevice= 5.04
[    2.643572] usb usb4: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    2.644218] usb usb4: Product: Generic Platform OHCI controller
[    2.644748] usb usb4: Manufacturer: Linux 5.4.0-rc1-rockchip64 ohci_hcd
[    2.645338] usb usb4: SerialNumber: fe3e0000.usb
[    2.646188] hub 4-0:1.0: USB hub found
[    2.646558] hub 4-0:1.0: 1 port detected
[    2.647996] usbcore: registered new interface driver usb-storage
[    2.651737] i2c /dev entries driver
[    2.654950] rk808 0-001b: chip id: 0x0
[    2.658745] rk808-regulator rk808-regulator: there is no dvs0 gpio
[    2.659321] rk808-regulator rk808-regulator: there is no dvs1 gpio
[    2.659947] DCDC_REG1: supplied by vcc3v3_sys
[    2.661214] DCDC_REG2: supplied by vcc3v3_sys
[    2.662136] DCDC_REG3: supplied by vcc3v3_sys
[    2.662794] DCDC_REG4: supplied by vcc3v3_sys
[    2.663779] LDO_REG1: supplied by vcc3v3_sys
[    2.665410] LDO_REG2: supplied by vcc3v3_sys
[    2.666751] LDO_REG3: supplied by vcc3v3_sys
[    2.668239] LDO_REG4: supplied by vcc3v3_sys
[    2.669714] LDO_REG5: supplied by vcc3v3_sys
[    2.671040] LDO_REG6: supplied by vcc3v3_sys
[    2.672531] LDO_REG7: supplied by vcc3v3_sys
[    2.673863] LDO_REG8: supplied by vcc3v3_sys
[    2.675364] SWITCH_REG1: supplied by vcc3v3_sys
[    2.676042] SWITCH_REG2: supplied by vcc3v3_sys
[    2.679851] rk808-rtc rk808-rtc: registered as rtc0
[    2.680987] fan53555-regulator 0-0040: FAN53555 Option[8] Rev[1] Detected!
[    2.681629] fan53555-reg: supplied by vcc3v3_sys
[    2.683134] vcc1v8_s3: supplied by vcc_1v8
[    2.684218] fan53555-regulator 0-0041: FAN53555 Option[8] Rev[1] Detected!
[    2.684850] fan53555-reg: supplied by vcc3v3_sys
[    2.696261] sdhci: Secure Digital Host Controller Interface driver
[    2.696810] sdhci: Copyright(c) Pierre Ossman
[    2.697200] Synopsys Designware Multimedia Card Interface Driver
[    2.698215] dwmmc_rockchip fe320000.dwmmc: IDMAC supports 32-bit address mode.
[    2.698871] dwmmc_rockchip fe320000.dwmmc: Using internal DMA controller.
[    2.699472] dwmmc_rockchip fe320000.dwmmc: Version ID is 270a
[    2.700000] dwmmc_rockchip fe320000.dwmmc: DW MMC controller at irq 25,32 bit host data width,256 deep fifo
[    2.700914] dwmmc_rockchip fe320000.dwmmc: Got CD GPIO
[    2.707063] random: fast init done
[    2.714065] mmc_host mmc0: Bus speed (slot 0) = 400000Hz (slot req 400000Hz, actual 400000HZ div = 0)
[    2.728272] sdhci-pltfm: SDHCI platform and OF driver helper
[    2.729899] mmc1: CQHCI version 5.10
[    2.754768] mmc1: SDHCI controller on fe330000.sdhci [fe330000.sdhci] using ADMA
[    2.757632] ledtrig-cpu: registered to indicate activity on CPUs
[    2.759826] usbcore: registered new interface driver usbhid
[    2.760375] usbhid: USB HID core driver
[    2.769076] xt_time: kernel timezone is -0000
[    2.769478] IPVS: Registered protocols (TCP, UDP, SCTP, AH, ESP)
[    2.770019] IPVS: Connection hash table configured (size=4096, memory=64Kbytes)
[    2.770744] IPVS: ipvs loaded.
[    2.771015] IPVS: [rr] scheduler registered.
[    2.771390] IPVS: [wrr] scheduler registered.
[    2.771772] IPVS: [lc] scheduler registered.
[    2.772147] IPVS: [wlc] scheduler registered.
[    2.772529] IPVS: [fo] scheduler registered.
[    2.772903] IPVS: [ovf] scheduler registered.
[    2.773288] IPVS: [lblc] scheduler registered.
[    2.773680] IPVS: [lblcr] scheduler registered.
[    2.774077] IPVS: [dh] scheduler registered.
[    2.774451] IPVS: [sh] scheduler registered.
[    2.774830] IPVS: [sed] scheduler registered.
[    2.775212] IPVS: [nq] scheduler registered.
[    2.775641] Initializing XFRM netlink socket
[    2.776046] NET: Registered protocol family 17
[    2.776607] 8021q: 802.1Q VLAN Support v1.8
[    2.777044] 9pnet: Installing 9P2000 support
[    2.777442] Key type dns_resolver registered
[    2.778133] registered taskstats version 1
[    2.778494] Loading compiled-in X.509 certificates
[    2.780390] zswap: default zpool zbud not available
[    2.780824] zswap: pool creation failed
[    2.781680] Btrfs loaded, crc32c=crc32c-generic
[    2.796211] mmc_host mmc0: Bus speed (slot 0) = 50000000Hz (slot req 50000000Hz, actual 50000000HZ div = 0)
[    2.797185] mmc0: new high speed SDHC card at address aaaa
[    2.798309] mmcblk0: mmc0:aaaa SC16G 14.8 GiB
[    2.803233] rk808-rtc rk808-rtc: setting system clock to 2013-01-18T08:56:29 UTC (1358499389)
[    2.803987] of_cfs_init
[    2.804222] of_cfs_init: OK
[    2.804975] ALSA device list:
[    2.805238]   No soundcards found.
[    2.805673]  mmcblk0: p1
[    2.805872] Freeing unused kernel memory: 896K
[    2.822660] Run /init as init process
[    2.846661] usb 2-1: new high-speed USB device number 2 using ehci-platform
[    2.995572] rk_gmac-dwmac fe300000.ethernet: IRQ eth_wake_irq not found
[    2.996184] rk_gmac-dwmac fe300000.ethernet: IRQ eth_lpi not found
[    2.996890] rk_gmac-dwmac fe300000.ethernet: PTP uses main clock
[    2.997562] rk_gmac-dwmac fe300000.ethernet: clock input or output? (input).
[    2.998195] rk_gmac-dwmac fe300000.ethernet: TX delay(0x28).
[    2.998746] rk_gmac-dwmac fe300000.ethernet: RX delay(0x11).
[    2.999256] rk_gmac-dwmac fe300000.ethernet: integrated PHY? (no).
[    2.999859] rk_gmac-dwmac fe300000.ethernet: cannot get clock clk_mac_speed
[    3.000475] rk_gmac-dwmac fe300000.ethernet: clock input from PHY
[    3.003490] usb 2-1: New USB device found, idVendor=1a40, idProduct=0101, bcdDevice= 1.00
[    3.004217] usb 2-1: New USB device strings: Mfr=0, Product=1, SerialNumber=0
[    3.004843] usb 2-1: Product: USB 2.0 Hub [MTT]
[    3.005681] hub 2-1:1.0: USB hub found
[    3.006072] rk_gmac-dwmac fe300000.ethernet: init for RGMII
[    3.006114] hub 2-1:1.0: 4 ports detected
[    3.006855] rk_gmac-dwmac fe300000.ethernet: User ID: 0x10, Synopsys ID: 0x35
[    3.007569] rk_gmac-dwmac fe300000.ethernet:         DWMAC1000
[    3.008037] rk_gmac-dwmac fe300000.ethernet: DMA HW capability register supported
[    3.008701] rk_gmac-dwmac fe300000.ethernet: RX Checksum Offload Engine supported
[    3.009365] rk_gmac-dwmac fe300000.ethernet: COE Type 2
[    3.009829] rk_gmac-dwmac fe300000.ethernet: TX Checksum insertion supported
[    3.010913] rk_gmac-dwmac fe300000.ethernet: Wake-Up On Lan supported
[    3.011544] rk_gmac-dwmac fe300000.ethernet: Normal descriptors
[    3.012663] rk_gmac-dwmac fe300000.ethernet: Ring mode enabled
[    3.013195] rk_gmac-dwmac fe300000.ethernet: Enable RX Mitigation via HW Watchdog Timer
[    3.013914] rk_gmac-dwmac fe300000.ethernet: device MAC address 6e:87:d0:94:1f:50
[    3.021010] OF: graph: no port node found in /syscon@ff770000/usb2-phy@e450/otg-port
[    3.021702] dwc3 fe800000.dwc3: IRQ otg not found
[    3.022116] dwc3 fe800000.dwc3: IRQ dwc_usb3 not found
[    3.023212] dwc3 fe800000.dwc3: IRQ host not found
[    3.023642] dwc3 fe800000.dwc3: IRQ dwc_usb3 not found
[    3.024332] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    3.024827] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 5
[    3.025602] xhci-hcd xhci-hcd.0.auto: hcc params 0x0220fe64 hci version 0x110 quirks 0x0000000002010010
[    3.026437] xhci-hcd xhci-hcd.0.auto: irq 221, io mem 0xfe800000
[    3.027226] usb usb5: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 5.04
[    3.027952] usb usb5: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.028583] usb usb5: Product: xHCI Host Controller
[    3.029011] usb usb5: Manufacturer: Linux 5.4.0-rc1-rockchip64 xhci-hcd
[    3.029589] usb usb5: SerialNumber: xhci-hcd.0.auto
[    3.030325] hub 5-0:1.0: USB hub found
[    3.030688] hub 5-0:1.0: 1 port detected
[    3.031053] dwc3 fe900000.dwc3: IRQ host not found
[    3.031476] dwc3 fe900000.dwc3: IRQ dwc_usb3 not found
[    3.032164] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    3.032805] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 6
[    3.033577] xhci-hcd xhci-hcd.1.auto: hcc params 0x0220fe64 hci version 0x110 quirks 0x0000000002010010
[    3.034423] xhci-hcd xhci-hcd.1.auto: irq 222, io mem 0xfe900000
[    3.035143] usb usb6: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 5.04
[    3.035868] usb usb6: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.036499] usb usb6: Product: xHCI Host Controller
[    3.036927] usb usb6: Manufacturer: Linux 5.4.0-rc1-rockchip64 xhci-hcd
[    3.037505] usb usb6: SerialNumber: xhci-hcd.1.auto
[    3.038023] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    3.038668] hub 6-0:1.0: USB hub found
[    3.039020] hub 6-0:1.0: 1 port detected
[    3.039504] xhci-hcd xhci-hcd.1.auto: xHCI Host Controller
[    3.039992] xhci-hcd xhci-hcd.1.auto: new USB bus registered, assigned bus number 7
[    3.040667] xhci-hcd xhci-hcd.1.auto: Host supports USB 3.0 SuperSpeed
[    3.041261] usb usb7: We don't know the algorithms for LPM for this host, disabling LPM.
[    3.042008] usb usb7: New USB device found, idVendor=1d6b, idProduct=0003, bcdDevice= 5.04
[    3.042740] usb usb7: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.043371] usb usb7: Product: xHCI Host Controller
[    3.043799] usb usb7: Manufacturer: Linux 5.4.0-rc1-rockchip64 xhci-hcd
[    3.044377] usb usb7: SerialNumber: xhci-hcd.1.auto
[    3.045020] hub 7-0:1.0: USB hub found
[    3.045362] hub 7-0:1.0: 1 port detected
[    3.045811] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 8
[    3.046484] xhci-hcd xhci-hcd.0.auto: Host supports USB 3.0 SuperSpeed
[    3.047127] usb usb8: We don't know the algorithms for LPM for this host, disabling LPM.
[    3.047870] usb usb8: New USB device found, idVendor=1d6b, idProduct=0003, bcdDevice= 5.04
[    3.048593] usb usb8: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    3.049224] usb usb8: Product: xHCI Host Controller
[    3.049651] usb usb8: Manufacturer: Linux 5.4.0-rc1-rockchip64 xhci-hcd
[    3.050228] usb usb8: SerialNumber: xhci-hcd.0.auto
[    3.050808] hub 8-0:1.0: USB hub found
[    3.051149] hub 8-0:1.0: 1 port detected
[    3.111275] libphy: stmmac: probed
[    3.111598] RTL8211E Gigabit Ethernet stmmac-0:00: attached PHY driver [RTL8211E Gigabit Ethernet] (mii_bus:phy_addr=stmmac-0:00, irq=POLL)
[    3.112701] RTL8211E Gigabit Ethernet stmmac-0:01: attached PHY driver [RTL8211E Gigabit Ethernet] (mii_bus:phy_addr=stmmac-0:01, irq=POLL)
[    9.374724] phy phy-ff770000.syscon:usb2-phy@e450.2: charger = USB_DCP_CHARGER
[   16.183122] EXT4-fs (mmcblk0p1): mounted filesystem with writeback data mode. Opts: (null)
[   16.770628] systemd[1]: System time before build time, advancing clock.
[   16.830052] NET: Registered protocol family 10
[   16.832900] Segment Routing with IPv6
[   16.862085] systemd[1]: systemd 237 running in system mode. (+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN -PCRE2 default-hierarchy=hybrid)
[   16.864404] systemd[1]: Detected architecture arm64.
[   16.901908] systemd[1]: Set hostname to <roc>.
[   16.936989] ttyS2 - failed to request DMA
[   17.139387] systemd[1]: File /lib/systemd/system/systemd-journald.service:35 configures an IP firewall (IPAddressDeny=any), but the local system does not support BPF/cgroup based firewalling.
[   17.140886] systemd[1]: Proceeding WITHOUT firewalling in effect! (This warning is only shown for the first loaded unit using IP firewalling.)
[   17.172771] random: crng init done
[   17.240386] systemd[1]: /lib/systemd/system/rk3399-bluetooth.service:11: Support for option SysVStartPriority= has been removed and it is ignored
[   17.304728] systemd[1]: Started Forward Password Requests to Wall Directory Watch.
[   17.306076] systemd[1]: Created slice User and Session Slice.
[   17.306782] systemd[1]: Reached target Remote File Systems.
[   17.307412] systemd[1]: Reached target User and Group Name Lookups.
[   17.384969] EXT4-fs (mmcblk0p1): re-mounted. Opts: commit=600,errors=remount-ro
[   17.444723] systemd-journald[417]: Received request to flush runtime journal from PID 1

Armbian 19.11.3 Bionic ttyS2

roc login: root
Password:
Last login: Mon Dec  2 01:55:26 UTC 2019 on ttyS2
 ____   ___   ____      ____  _  ____________ ___   ___        ____   ____
|  _ \ / _ \ / ___|    |  _ \| |/ /___ /___ // _ \ / _ \      |  _ \ / ___|
| |_) | | | | |   _____| |_) | ' /  |_ \ |_ \ (_) | (_) |_____| |_) | |
|  _ <| |_| | |__|_____|  _ <| . \ ___) |__) \__, |\__, |_____|  __/| |___
|_| \_\\___/ \____|    |_| \_\_|\_\____/____/  /_/   /_/      |_|    \____|

Welcome to Armbian Bionic with Linux 5.4.0-rc1-rockchip64

System load:   1.48 0.38 0.13   Up time:       0 min
Memory usage:  7 % of 3868MB    IP:
CPU temp:      47°C
Usage of /:    13% of 15G

root@roc:~# uname -a
Linux roc 5.4.0-rc1-rockchip64 #19.11.3 SMP PREEMPT Mon Dec 2 01:49:54 UTC 2019 aarch64 aarch64 aarch64 GNU/Linux
```
